### ISSUE TITLE
Loss function of VAE

### ISSUE THREAD

**Issue by bcpenggh** [2017-fall-DL-training-program/VAE-GAN-and-VAE-GAN#11]
Dear TA,
In the VAE handout, the loss function is shown in item 1 of "Implementation Details." Sorry that I don't know how to paste a figure here. The 1st term of the loss function is to make the decoder output close to the encoder input. So I expect the log of probability should be as large as possible. On the other hand the 2nd term KL divergence is to make the encoder output close to the expected distribution, so the distance is as small as possible. In that case, the loss function gets larger and larger, instead of smaller and smaller. This seems different from the intuitive definition. I understand correctly?

**Comment by hui-po-wang** [comment]
Hi @bcpenggh,

The whole story is like this: we want to maximize the log-likelihood, but we can not maximize it directly because of the neural networks in the function, so we use variational inference to "indirectly" maximize the log-likelihood by maximizing ```variational lower bound```. That is the equation in the figure below and our handouts.

Again, our goal is to ```maximize``` the likelihood, so we should try to ```maximize``` its variational lower bound：

1. We find that the first term describes the probability distribution of the outputs from the decoder given the latent codes. If we model it as a Gaussian distribution and we want the first term to be as large as possible, the most intuitive way is to ```decrease``` the distance between the outputs (denoted it as ```b``` in the second figure) and the ground truth (i,e, the original inputs and denoted by ```x```), so it turns out that we try to ```minimize``` the mean square errors between reconstruction and original images.

2. In terms of the second term, ```maximizing``` the negative KL divergence equals ```minimizing``` it.

In conclusion, we try to ```maximize``` the equation in the first figure and our handouts, but it actually has the same effect that we ```minimize``` the KL divergence and the mean square error.

![screenshot from 2017-11-28 15-40-08](https://user-images.githubusercontent.com/10228478/33307705-7bad1d9a-d452-11e7-82d4-49c4813112fb.png)

This Gaussian is copied and pasted from Wiki, so the notation may be not so suitable for the problem.
![screenshot from 2017-11-28 15-53-26](https://user-images.githubusercontent.com/10228478/33308161-4bfcc51c-d454-11e7-9c42-0ebf8742a147.png)

Thanks
